### DISCUSSION TITLE
PSACanVanBRIDGE

### DISCUSSION THREAD

**Discussion by vitalik12347** [morcibacsi/PSAVanCanBridge#35] [Q&A]
Hello! How to purchase your PSAVanCanBridge for a 2003 Peugeot 307! I would really help out.Thanks

**Comment by morcibacsi** [comment]
Currently I don't have any spare boards, a new (more robust) hardware revision is under development, but it is going to take a few weeks until it is finished. Send an email to ![image](https://github.com/morcibacsi/PSAVanCanBridge/assets/6507173/d78b6aee-f717-4a63-b422-d3d281dd368d)

**Comment by manuel2462002** [comment]
Hello! Where can I find the updated software for PSAVanCanBridge Hardware v1.5?

**Reply by morcibacsi** [reply]
It is on the main branch.

**Comment by Lapik399** [comment]
Hi. Looking to upgrade my old Berlingo too with RD45. I was wondering, if it would be too complicated to add few functions to it- I'd like to have comfort lightning also on unlocking (if any gp inputs are available, I'd fit a switch to activate/deactivate the function). Is it possible to send such messages over VAN-bus V2C and overriding other modules?

**Reply by morcibacsi** [reply]
Hi, unfortunately we can't really alter anything regarding how the car works. Locks and lighting (and most things in the car) is managed by the BSI. Given that basically I am emulating a peripheral (the EMF display) of the BSI I can't inject new messages to alter behaviour. In theory some things might be possible, but what I wrote here is what and how could be done: https://github.com/morcibacsi/PSAVanCanBridge/issues/48#issuecomment-3493320951

The TL;DR version is this:

> given how the buses work it is impossible to "inject messages". Every peripheral sends their messages periodically. If you inject a different message it might work for a brief moment, but as soon as the original sender sends the message again, yours would have been overwritten. 

So a separate gateway could be written between the peripherals which are present on the picture at the above mentioned issue, but it is just too complicated to implement. It is really sad, I'd really love to add new features to our cars if it was possible easily. :(
